fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -517,17 +524,27 @@ public void setCurrentCatalog(String currentCatalog) {
     public void kill(boolean killConnection) {
         LOG.warn("kill timeout query, {}, kill connection: {}",
                 getMysqlChannel().getRemoteHostPortString(), killConnection);
+        // Now, cancel running process.
+        StmtExecutor executorRef = executor;
+        if (executorRef != null) {
+            executorRef.cancel();
+        }
+        while (closed == false) {
+            // if we don't wait for some time here, mysql client will reconnect
+            // previous session and rerun previous command automatically
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
```

Comment:
I suggest you quit the loop when a InterruptedException is caught. This exception is meant to interrupt the process.